web/components/manage-vault.tsx

```diff
@@ -291,7 +291,7 @@ export const ManageVault = ({ match }) => {
       functionName: 'burn',
       functionArgs: [
         uintCV(match.params.id),
-        uintCV(parseFloat(usdToBurn) * 1000000),
+        uintCV(parseFloat(usdToBurn - stabilityFee) * 1000000),
```

Comment:
The stabilityFee comes from the contract so the integer includes 6 digits. While usdToBurn comes from user input so does not include 6 digits. I think it needs to be: uintCV((parseFloat(usdToBurn) * 1000000) - stabilityFee)

As we are now subtracting the stability fee from user input we can also optimise the post condition a bit:
let totalToBurn = Number(usdToBurn) + 2 * (stabilityFee / 1000000);
Can become:
let totalToBurn = Number(usdToBurn) + (stabilityFee / 1000000);